### PR TITLE
Updating README to only python2, merged dependency installation

### PR DESCRIPTION
**Pull request by anandtrex**

### CODE REVIEW

File: examples/pattern_matching_showcase/README.md [hunk 1]
```diff
@@ -32,31 +32,52 @@ This guide assumes that you want to install everything into your local home fold
...
+pip install --upgrade matplotlib  # (Optional, If you want to have nice plotting, you should upgrade `matplotlib` to the newest version)
 ```
 
+Notes:
+
+* Use `pip install --user <package name>` to install the python packages locally
+* Use `sudo pip install <package name>` to install the python packages globally
+* The python packages can also be installed in a [virtualenv](https://virtualenv.pypa.io/en/stable/).
+* For python3, use `pip3` to install python packages and install the python3 versions of the packages for NEST dependencies. 
+* Python3 is NOT officially supported by this package, but has been known to work occasionally.
```
Comment (mhoff) [inline examples/pattern_matching_showcase/README.md]: Shouldn't these notes go above the installation instructions?

File: examples/pattern_matching_showcase/README.md [hunk 2]
```diff
@@ -32,31 +32,52 @@ This guide assumes that you want to install everything into your local home fold
 ## Preparation
 
 Add the following lines to your `~/.bashrc` (or `~/.zshrc` or `~/.profile` etc.)
+
 ```bash
 export TARGET_DIR=$HOME/opt/
```
Comment (mhoff) [inline examples/pattern_matching_showcase/README.md]: `$NUM_CORES` never defined.

File: examples/pattern_matching_showcase/README.md (hunk 1 above)
Comment (mhoff) [inline examples/pattern_matching_showcase/README.md]: Okay, looking at the rendered README it is visually close enough, isn't it?

File: examples/pattern_matching_showcase/README.md (hunk 2 above)
Comment (anandtrex) [inline examples/pattern_matching_showcase/README.md]: Fixed.

File: examples/pattern_matching_showcase/README.md (hunk 1 above)
Comment (anandtrex) [inline examples/pattern_matching_showcase/README.md]: Fixed.

**Pull request merged**